infra/Pos/Communication/Protocol.hs

```diff
@@ -253,30 +225,6 @@ type OnNewSlotComm ctx m =
     , HasConfiguration
     )
 
-onNewSlot'
-    :: OnNewSlotComm ctx m
-    => Bool -> (SlotId -> WorkerSpec m, outSpecs) -> (WorkerSpec m, outSpecs)
-onNewSlot' startImmediately (h, outs) =
-    (,outs) . ActionSpec $ \vI sA ->
-        onNewSlot startImmediately $
-            \slotId -> let ActionSpec h' = h slotId
-                        in h' vI sA
-onNewSlotWorker
-    :: OnNewSlotComm ctx m
-    => Bool -> OutSpecs -> (SlotId -> Worker m) -> (WorkerSpec m, OutSpecs)
-onNewSlotWorker b outs = onNewSlot' b . workerHelper outs
-
-localOnNewSlotWorker
-    :: LocalOnNewSlotComm ctx m
-    => Bool -> (SlotId -> m ()) -> (WorkerSpec m, OutSpecs)
-localOnNewSlotWorker b h = (ActionSpec $ \__vI __sA -> onNewSlot b h, mempty)
-
-localWorker :: m () -> (WorkerSpec m, OutSpecs)
-localWorker = localSpecs
-
-localSpecs :: m a -> (ActionSpec m a, OutSpecs)
-localSpecs h = (ActionSpec $ \__vI __sA -> h, mempty)
-
```

Comment:
This stuff moved to `cardano-sl-lib` because `Action m`/`Worker m` depend upon a `Diffusion m`, not `SendActions`, and `Diffusion m` is defined using types from the various intermediate packages (txp, update, ssc), which themselves depend upon infra. This is well-motivated: those packages shouldn't be defining workers, since workers are part of an application.